File: Reviewers/Entity/Merchandise.swift
import Foundation

struct Merchandise: Identifiable, Hashable {
    let id: String
    let enable: Bool
    let status: MerchandiseStatus
    let name: String
    let code: String
    let codeType: CodeType
    let image: String
    let imageReferenceReviewId: String
    let createdAt: Date
    let updatedAt: Date
}


enum MerchandiseStatus: String {
    case waitingForReview = "WaitingForReview"
    case reviewComplete = "ReviewCompleted"
}

File: admin/entity/database/merchandise.go
package database

import "time"

type Merchandise struct {
	Id        string
	CreatedAt time.Time `firestore:"createdAt,serverTimestamp"`
	Enable    bool      `firestore:"enable"`
	Name      string    `firestore:"name"`
	Status    string    `firestore:"status"`
	UpdatedAt time.Time `firestore:"updatedAt,serverTimestamp"`
}

//func (m *Merchandise) Data() map[string]interface{} {
//	return map[string]interface{}{
//		"createdAt": m.CreatedAt,
//		"enable":    m.Enable,
//		"name":      m.Name,
//		"status":    m.Status,
//		"updatedAt": m.UpdatedAt,
//	}
//}
